x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/TransportPutFollowAction.java

```diff
@@ -211,6 +244,29 @@ private void initiateFollowing(
         ));
     }
 
+    static DataStream updateLocalDataStream(Index backingIndexToFollow,
+                                            DataStream localDataStream,
+                                            DataStream remoteDataStream) {
+        if (localDataStream == null) {
+            // The data stream and the backing indices have been created and validated in the remote cluster,
+            // just copying the data stream is in this case safe.
+            return new DataStream(remoteDataStream.getName(), remoteDataStream.getTimeStampField(),
+                List.of(backingIndexToFollow), remoteDataStream.getGeneration(), remoteDataStream.getMetadata());
+        } else {
+            List<Index> backingIndices = new ArrayList<>(localDataStream.getIndices());
+            backingIndices.add(backingIndexToFollow);
+
+            // When following an older backing index it should be positioned before the newer backing indices.
+            // Currently the assumption is that the newest index (highest generation) is the write index.
+            // (just appending an older backing index to the list of backing indices would break that assumption)
+            // (string sorting works because of the naming backing index naming scheme)
+            backingIndices.sort(Comparator.comparing(Index::getName));
+
+            return new DataStream(localDataStream.getName(), localDataStream.getTimeStampField(), backingIndices,
```

Comment:
What if the generation of the local datastream is higher than the remote?

Reply:
The generation isn't really used yet, so there isn't a real consequence when the local data stream's generation is higher than the leader's data stream. 

This would happen if the local data stream would be rollover independently. This is possible, but shouldn't really happen. Because that would create an index that doesn't follow anything and will remain empty and unused. The rollover should only occur in the remote cluster.

So I think we should try to prevent data streams from being rolled over when that data stream is actually just following another data stream. What I think we can do for this is add a flag to a data stream that it is replicated and a rollover would disallow rolling over a replicated data stream. When unfollowing that data stream, the replicate flag would be unset. What do you think about this?

Reply:
> So I think we should try to prevent data streams from being rolled over when that data stream is actually just following another data stream. What I think we can do for this is add a flag to a data stream that it is replicated and a rollover would disallow rolling over a replicated data stream. When unfollowing that data stream, the replicate flag would be unset. What do you think about this?

+1. This makes a lot of sense.

Reply:
Martijn will implement this in a follow up.

Reply:
Yes, in a followup PR, I will add an additional field to `DataStream` class that indicates that it is a replicated data stream. Then in the rollover api validation will be added that prevents rolling over a replicated data stream.